### PR TITLE
community/knot-resolver: upgrade to 4.2.2

### PR DESCRIPTION
**Pull request by tcely**

### CODE REVIEW

File: community/knot-resolver/APKBUILD [hunk 1]
```diff
@@ -16,17 +16,18 @@ depends="dns-root-hints dnssec-root lua5.1-sec lua5.1-socket"
...
 depends_dev="knot-dev>=2.8.0 libedit-dev libuv-dev>=1.7 lmdb-dev luajit-dev>=2.0 $_depends_dnstap_dev"
+depends_static="$pkgname-dev"
 makedepends="$depends_dev bash cmake gnutls-dev luacheck meson>=0.46 ninja pkgconf py3-flake8"
 checkdepends="cmocka-dev"
 install="$pkgname.pre-install"
-subpackages="$pkgname-mod-http:http:noarch $pkgname-mod-dnstap:dnstap $pkgname-dev $pkgname-doc $pkgname-openrc"
+subpackages="$pkgname-mod-http:http:noarch $pkgname-mod-dnstap:dnstap $pkgname-libs-static $pkgname-dev $pkgname-doc $pkgname-openrc"
 source="
-	https://secure.nic.cz/files/${_pkgname}/${_pkgname}-$pkgver.tar.xz
+	https://secure.nic.cz/files/${_upstream}/${_upstream}-$pkgver.tar.xz
```
Comment (maxice8) [inline community/knot-resolver/APKBUILD]: @Ikke asked to use the written out name as noted [here](https://github.com/maxice8/atools/blob/master/alint.5.scd#L316)

File: main/lmdb/APKBUILD [hunk 2]
```diff
@@ -2,7 +2,7 @@
 # Maintainer: Natanael Copa <ncopa@alpinelinux.org>
 pkgname=lmdb
 pkgver=0.9.24
-pkgrel=0
+pkgrel=1
```
Comment (ncopa) [inline main/lmdb/APKBUILD]: Why is this needed? (see 50e76a2dd93dff1d8d3b59616a822459fbfd08d7)

If the problem is that there is no `provides=pc:lmdb`, then why would a simple rebuild fix it?

File: main/luajit/APKBUILD [hunk 3]
```diff
@@ -4,7 +4,7 @@ pkgname=luajit
 _pkgname=LuaJIT
 pkgver=2.1.0_beta3
 _pkgver=${pkgver/_/-}
-pkgrel=4
+pkgrel=5
```
Comment (ncopa) [inline main/luajit/APKBUILD]: why is this needed?

Comment (tcely) [reply main/luajit/APKBUILD]: Because `abuild` is using `pkg-config --modversion` for package versions and sometimes that value is not legal for use with `apk`.

File: main/lmdb/APKBUILD (hunk 2 above)
Comment (tcely) [reply main/lmdb/APKBUILD]: https://github.com/alpinelinux/aports/blob/db01a69b12d719e2c8ff643694499e38e5e9fb19/main/lmdb/APKBUILD#L41

Turns out this was the cause.

**Comment by tcely** [comment]
Rebuilding `knot` is needed until https://github.com/alpinelinux/aports/issues/11811 is merged.

**Comment by Ikke** [comment]
> Rebuilding `knot` is needed until #11811 is merged.

Done
